### PR TITLE
Unpredictible number reference in java

### PR DESCRIPTION
**Pull request by jot2re**

### CODE REVIEW

**Comment by colourful-land** [comment]
I'm asking @fyang1024 to work directly on the same branch his JavaScript implementation, so if changes needed before merge they can be applied to both JS and Java version.

**Review by colourful-land — CHANGES_REQUESTED** [review]
NFTOFFER.asn doesn't belong to this branch, kindly remove it and consolidate (git-rebase) if you feel the need.

**Comment by jot2re** [comment]
Sorry about the extra file. I must accidentally have based this branch on another branch I was working on. The file is removed now.

File: src/main/java/org/devcon/ticket/UnpredictableNumberTool.java [hunk 1]
```diff
@@ -0,0 +1,81 @@
...
+  public UnpredictableNumberBundle getUnpredictableNumberBundle() {
+    long expiration = Clock.systemUTC().millis() + validityInMs;
+    return new UnpredictableNumberBundle(getUnpredictableNumber(expiration), domain, expiration);
+  }
+
+  private String getUnpredictableNumber(long expirationInMs) {
+    hmac.reset();
+    hmac.update(longToBytes(expirationInMs), 0, Long.BYTES);
+    hmac.update(domain.getBytes(StandardCharsets.UTF_8), 0, domain.getBytes(StandardCharsets.UTF_8).length);
+    byte[] digest = new byte[256 / 8];
```
Comment (oleggrib) [inline src/main/java/org/devcon/ticket/UnpredictableNumberTool.java:57]: Can you  generate random number , add it to hmac and prepend that UN with that number?

Like that logic:
expiry: now + 3600
UN1: rand()
UN2: hmac(concat(UN1, expiry), bogotá-secret)
UN: concat(UN1, UN2)

Comment (jot2re) [reply src/main/java/org/devcon/ticket/UnpredictableNumberTool.java:68]: I have added randomness (and kept the domain name).

**Comment by jot2re** [comment]
@colourful-land please check if it is ok now.

**Review by oleggrib — APPROVED** [review]

**Pull request merged**